sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.cpp

```diff
@@ -2,12 +2,130 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "flutter/sky/engine/platform/fonts/fuchsia/FontCacheFuchsia.h"
+
+#include <limits>
+#include <magenta/syscalls.h>
+#include <utility>
+
+#include "flutter/sky/engine/platform/fonts/AlternateFontFamily.h"
 #include "flutter/sky/engine/platform/fonts/FontCache.h"
+#include "flutter/sky/engine/platform/fonts/FontDescription.h"
+#include "lib/ftl/logging.h"
+#include "mojo/services/ui/fonts/interfaces/font_provider.mojom.h"
+#include "third_party/skia/include/core/SkData.h"
+#include "third_party/skia/include/core/SkTypeface.h"
+#include "third_party/skia/include/ports/SkFontMgr.h"
 
 namespace blink {
+namespace {
+
+uint32_t ToMojoWeight(FontWeight weight) {
+  switch (weight) {
+    case FontWeight100:
+      return 100;
+    case FontWeight200:
+      return 200;
+    case FontWeight300:
+      return 300;
+    case FontWeight400:
+      return 400;
+    case FontWeight500:
+      return 500;
+    case FontWeight600:
+      return 600;
+    case FontWeight700:
+      return 700;
+    case FontWeight800:
+      return 800;
+    case FontWeight900:
+      return 900;
+  }
+  ASSERT_NOT_REACHED();
+  return 400;
+}
+
+mojo::FontSlant ToMojoSlant(FontStyle style) {
+  switch (style) {
+    case FontStyleNormal:
+      return mojo::FontSlant::UPRIGHT;
+    case FontStyleItalic:
+      return mojo::FontSlant::ITALIC;
+  }
+  ASSERT_NOT_REACHED();
+  return mojo::FontSlant::UPRIGHT;
+}
+
+void UnmapMemory(const void* buffer, void* context) {
+  mx_process_unmap_vm(0, reinterpret_cast<uintptr_t>(buffer), 0);
```

Comment:
I asked Travis, and he says that passing zero will remove the whole mapping.  I added that to the docs for this call: https://fuchsia.googlesource.com/magenta/+/master/docs/syscalls/process_unmap_vm.md